src/wasm-js.cpp

```diff
@@ -417,7 +417,7 @@ extern "C" void EMSCRIPTEN_KEEPALIVE instantiate() {
     void store(Store* store_, Address address, Literal value) override {
       uint32_t addr = address;
       // support int64 stores
-      if (value.type == WasmType::i64) {
+      if (value.type == WasmType::i64 && store_->bytes > 4) {
```

Comment:
My thought was to futureproof; if we do something weird like i64.store48 we possibly want it to fall back to that logic.
Although that would still cause problems by writing 2 too many bytes, and it doesn't protect against anything larger, and overall yeah it's probably better to err on readability here.